Align API-Testing-Guidelines with new Enhancements and Alignments in Errors (#343)
camaraproject/Commonalities: documentation/API-Testing-Guidelines.md

### Error scenarios

* Test plan must be aligned with API specification regarding the coverage of HTTP statuses in the responses.
  - All errors explicitly documented in the API spec must be covered by one or more dedicated error scenarios.

>   - On the contrary, HTTP Statuses which are not explicitly documented in the API specification should not be covered in the test plan. If any of the non-mandatory "status" at Commonalities level is relevant in the context of a given API and deserves to have a specific test, it must be documented as well in the API specification.

Reviewer: I propose to rephrase a bit, and directly reference to Design Guidelines:

```suggestion
  - On the contrary, HTTP Statuses which are not explicitly documented in the API specification should not be covered in the test plan. If any of the non-mandatory "status" in API Design Guidelines is relevant in the context of a given API and deserves to have a specific test, it needs to be documented in the API specification as well.
```
